application/models/message.go

```diff
@@ -183,6 +185,90 @@ func (m *Message) Create(tx *pop.Connection, user User, requestUUID string, thre
 	return nil
 }
 
+// Todo Once gql is no longer supported, simplify the name of this function and get rid of the Create() function above
+// Create a new message if authorized.
+func (m *Message) CreateForRest(tx *pop.Connection, user User, input api.MessageInput) *api.AppError {
+	var request Request
+	if err := request.FindByUUID(tx, input.RequestID); err != nil {
+		appError := api.AppError{
+			Category: api.CategoryUser,
+			Key:      api.MessageBadRequestUUID,
+			Err:      errors.New("error with new message: " + err.Error()),
+		}
+		return &appError
+	}
+
+	if isVisible, err := request.IsVisible(tx, user); err != nil {
+		appError := api.AppError{
+			Category: api.CategoryInternal,
+			Key:      api.QueryFailure,
+			Err:      errors.New("error with new message: " + err.Error()),
+		}
+		return &appError
+	} else if !isVisible {
+		appError := api.AppError{
+			Category: api.CategoryForbidden,
+			Key:      api.MessageRequestNotVisible,
+			Err:      errors.New("user cannot create a message on request"),
+		}
+		return &appError
+	}
+
+	var thread Thread
+	if input.ThreadID != nil && *input.ThreadID != "" {
```

Comment:
Out of curiosity, why it is necessary to check for both non-nil and non-empty string here?

Reply:
I guess there's a chance a non-nil but blank ThreadID could come through, in which case we want to ignore it.

If you try to just use the second condition on a nil pointer (without the first condition), it will panic.